src/main/java/homework_4/pages/MetalAndColorsPage.java

```diff
@@ -0,0 +1,137 @@
+
+/**
+ * Created by Dinara Shabanova on 28.11.2019.
+ */
+public class MetalAndColorsPage {
+
+    public MetalAndColorsPage() {
+        page(this);
+        unselectDefaultItem();
+    }
+
+    @FindBy(id = "summary-block")
+    private WebElement summary;
+
+    @FindBy(id = "elements-block")
+    private WebElement elements;
+
+    @FindBy(id = "colors")
+    private WebElement colors;
+
+    @FindBy(xpath = "//div[@id='metals']//span[@class='caret']")
+    private WebElement metals;
+
+    @FindBy(id = "salad-dropdown")
+    private WebElement vegetables;
+
+    @FindBy(xpath = "//div[@id='mCSB_1_container']//div[contains(text(),'Result')]")
+    private WebElement result;
+
+    public void chooseSummary(String item) {
+        colors.findElement(By.xpath("//label[@for='" + item + "']")).click();
+    }
+
+    public void chooseElements(String item) {
+        colors.findElement(By.xpath("//label[@for='" + item + "']")).click();
+    }
+
+    public void chooseMetals(String item) {
+        colors.findElement(By.xpath("//span[@class='text'][contains(text(),'" + item + "')]")).click();
+    }
+
+    public void chooseVegetables(String item) {
+        vegetables.findElement(By.xpath("//label[@for='" + item + "']")).click();
+    }
+
+    public void chooseColors(String item) {
+        colors.findElement(By.xpath("//span[contains(text(),'" + item + "')]")).click();
+    }
+
+    public void click(Entity entity) {
+        switch (entity) {
+            case COLORS:
+                colors.findElement(By.tagName("button")).click();
+                break;
+            case METALS:
+                metals.click();
+                break;
+            case SUBMIT:
+                $(By.id("submit-button")).click();
+                break;
+            case VEGETABLES_ENTITY:
+                vegetables.findElement(By.tagName("button")).click();
+                break;
+        }
+    }
+
+    private void unselectDefaultItem() {
+        $(By.id("salad-dropdown")).click();
+        if ($(By.xpath("//div[@id='salad-dropdown']//button[@type='button']"))
+                .getText().contains("Vegetable")) {
+            $(By.xpath("//div[@id='salad-dropdown']//label[@for='g7']")).click();
+        }
+    }
+
+    public Result getActualResult() {
+        int summary = 0;
+        String color = null;
+        String metals = null;
+        List<String> elements = null;
+        List<String> vegetables = new ArrayList<>();
+
+        List<WebElement> listResult = $(By.xpath("//ul[@class='panel-body-list results']")).findElements(By.tagName("li"));
+        for (WebElement el : listResult) {
+            if (el.getText().contains("Summary")) {
+                summary = Integer.parseInt(getValueFromActualResult(el.getText()).get(0).trim());
+            } else if (el.getText().contains("Color")) {
+                color = getValueFromActualResult(el.getText()).get(0).trim();
+            } else if (el.getText().contains("Metal")) {
```

Comment:
Possible `NullPointerException`. It will be better to swap **string** and `el`. `"Metal".equals(el.getText())`

Reply:
fix